### PR TITLE
goldfish: sync with qemu do rpmsg_syslog init at goldfish

### PR DESCRIPTION
**Pull request by GUIDINGLI**
## Summary

goldfish: sync with qemu do rpmsg_syslog init at goldfish

## Impact

goldfish

## Testing

goldfish

### CODE REVIEW

File: arch/arm/src/goldfish/goldfish_boot.c [hunk 1]
```diff
@@ -41,6 +41,16 @@
...
 #endif
 
+#include <nuttx/syslog/syslog_rpmsg.h>
+
+/****************************************************************************
+ * Private Data
+ ****************************************************************************/
+
+#ifdef CONFIG_SYSLOG_RPMSG
+static char g_syslog_rpmsg_buf[4096];
```
Comment (acassis) [inline arch/arm/src/goldfish/goldfish_boot.c:51]: Maybe the log buffer should be configurable, i.e. using CONFIG_SYSLOG_BUFSIZE

**Comment by nuttxpr** [comment]
[**\[Experimental Bot, please feedback here\]**](https://github.com/search?q=repo%3Aapache%2Fnuttx+13552&type=issues)



**No, this PR does not meet the NuttX requirements.**

**Missing Information:**

* **Summary:** 
    * Lacks a clear explanation of why the change is necessary. 
    * Needs to describe what "rpmsg_syslog init" is and why it needs synchronization with QEMU.
    * Should specify the functional area of the code being modified.
* **Impact:** 
    * Simply stating "goldfish" is insufficient. Explain what "goldfish" refers to and the specific impact of the change on it. 
    * Address all other impact categories (user, build, hardware, documentation, security, compatibility). 
* **Testing:**
    * Provide details about the build host (OS, CPU, compiler) used for testing.
    * Specify the exact goldfish target (board:config) used.
    * Include relevant testing logs from both before and after the change, demonstrating the issue and the fix.

**Recommendations:**

1. **Expand the Summary:** Clearly articulate the problem, the proposed solution, and the code sections involved.
2. **Complete the Impact Assessment:**  Address all impact categories, providing specific details where applicable. 
3. **Provide Comprehensive Testing Information:**  Include build host and target details and relevant testing logs. 

By providing this missing information, you will create a more informative and complete PR that adheres to NuttX's contribution guidelines.

File: arch/arm/src/goldfish/goldfish_boot.c (hunk 1 above)
Comment (CV-Bowen) [reply arch/arm/src/goldfish/goldfish_boot.c:51]: @acassis The buffer is provided by chip, should not use a system config to config this buffer size?

Comment (xiaoxiang781216) [reply arch/arm/src/goldfish/goldfish_boot.c:51]: CONFIG_SYSLOG_BUFSIZE is a syslog line buffer length, not suitable here.

Comment (acassis) [reply arch/arm/src/goldfish/goldfish_boot.c:51]: @xiaoxiang781216 true! Thanks for warning my mistake

**Pull request merged**